test/framework/github.py

```diff
@@ -1317,26 +1318,96 @@ def test_github_create_test_report(self):
                 'log_file': logfile,
             }),
         ]
+        environ = {
+            'USER': 'test',
+            'DONT_REMOVE_ME': 'test-123',
+        }
+        JWT_HDR = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9'
+        JWT_PLD = 'eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNzA4MzQ1MTIzLCJleHAiOjE3MDgzNTUxMjN9'
+        JWT_SIG = 'SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c'
+        secret_environ = {
+            'REMOVEME': 'test-abc',
+            'ReMoVeMe2': 'TeSt-xyz',
+
+            'AWS_ACCESS_KEY': 'AKIAIOSFODNN7EXAMPLE',
+            'AWS_SECRET_KEY': 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY',
+            'JWT': '.'.join([JWT_HDR, JWT_PLD, JWT_SIG]),
+            'GH_TOKEN': 'ghp_123456789_ABCDEFGHIJKlmnopqrstuvwxyz',
+            'SLACK_TOKEN': 'xoxb-1234567890-1234567890123-ABCDEFabcdef',
+
+            'API_SOMETHING': '1234567890',
+            'MY_PASSWORD': '1234567890',
+            'ABC_TOKEN': '1234567890',
+            'AUTH_XXX': '1234567890',
+            'LICENSE': '1234567890',
+            'WORLD_KEY': '1234567890',
+            'PRIVATE_INFO': '1234567890',
+            'SECRET_SECRET': '1234567890',
+            'INFO_CREDENTIALS': '1234567890',
+        }
         init_session_state = {
             'easybuild_configuration': ['EASYBUILD_DEBUG=1'],
-            'environment': {'USER': 'test'},
+            'environment': {**environ, **secret_environ},
             'module_list': [{'mod_name': 'test'}],
             'system_info': {'name': 'test'},
             'time': gmtime(0),
         }
+
+        # Test exclude_env_from_report_add/clear
+        exclude_env_from_report_add('REMOVEME'.lower())  # Also check that the name is uppercased in the check
+
         res = create_test_report("just a test", ecs_with_res, init_session_state)
         patterns = [
             "**SUCCESS** _test.eb_",
             "**FAIL (build issue)** _fail.eb_",
             "01 Jan 1970 00:00:00",
             "EASYBUILD_DEBUG=1",
+            "DONT_REMOVE_ME = test-123",
         ]
         for pattern in patterns:
             self.assertIn(pattern, res['full'])
 
-        for pattern in patterns[:2]:
+        # Test that excluded patterns works by matching also partial strings
+        exclude_patterns1 = [
+            'REMOVEME',
+            'ReMoVeMe2',
+        ]
+        # Test that known token regexes for ENV vars are excluded by default
+        exclude_patterns2 = [
+            'AWS_ACCESS_KEY',
+            'AWS_SECRET_KEY',
+            'JWT',
+            'GH_TOKEN',
```

Comment:
```suggestion
            'GH_TOKEN',
            'SAME_THING_DIFFERENT_NAME',
```

Reply:
Fixed with https://github.com/easybuilders/easybuild-framework/pull/4877/commits/39a6bf45a787adbb50c7edb3a495204d0c9de3ec to check for a group of variables based on value and another based on name